### PR TITLE
Fixes #20387 - Clean the redundant code for auto complete search

### PR DESCRIPTION
**Pull request by rabajaj0509**
This commit shall improve the usability by removing the extra spaces prefixed to the search string.

### CODE REVIEW

File: app/models/discovery_rule.rb [hunk 1]
```diff
@@ -26,6 +27,10 @@ class DiscoveryRule < ApplicationRecord
   scoped_search :on => :enabled
   scoped_search :relation => :hostgroup, :on => :name, :complete_value => true, :rename => :hostgroup
 
+  def remove_extra_spaces
+     self.search = search.strip
```
Comment (ohadlevy) [inline app/models/discovery_rule.rb]: how about to fix it on the search autocompleter, so its fixed for all screens that use search?

**Comment by rabajaj0509** [comment]
related pr : https://github.com/theforeman/foreman/pull/4693

**Comment by lzap** [comment]
Tests finally fixed [test].

**Comment by lzap** [comment]
Yeah we had it there because `model_of_controller` was not yet available in core, we added it later.

**Comment by rabajaj0509** [comment]
Not sure why the tests are failing here, when i see the test results it says - `no failures`.

**Comment by lzap** [comment]
Looks like an issue with CI - "Step ‘Publish JUnit test result report’ failed: no workspace left". One more try and then I will merge [test]

**Comment by lzap** [comment]
I can confirm search works for both discovered index and rule edit form, but I still see spaces in the dropdown. When you save or confirm the search, the extra space is trimmed tho. Merging thanks.

**Pull request merged**